packages/component/src/BasicSendBox.js

```diff
@@ -28,28 +29,46 @@ const DICTATION_INTERIMS_CSS = css({ flex: 10000 });
 const MICROPHONE_BUTTON_CSS = css({ flex: 1 });
 const TEXT_BOX_CSS = css({ flex: 10000 });
 
-const BasicSendBox = ({ className, dictationStarted, styleSet, webSpeechPonyfill }) => (
-  <div className={classNames(styleSet.sendBox + '', ROOT_CSS + '', className + '')} role="form">
-    <TypingIndicator />
-    <ConnectivityStatus />
-    <SuggestedActions />
-    <div className="main">
-      {!styleSet.options.hideUploadButton && <UploadButton />}
-      {dictationStarted ? (
-        <DictationInterims className={DICTATION_INTERIMS_CSS + ''} />
-      ) : (
-        <TextBox className={TEXT_BOX_CSS + ''} />
-      )}
-      <div>
-        {(webSpeechPonyfill || {}).SpeechRecognition ? (
-          <MicrophoneButton className={MICROPHONE_BUTTON_CSS + ''} />
+// TODO: [P3] We should consider exposing core/src/definitions and use it instead
+function activityIsSpeakingOrQueuedToSpeak({ channelData: { speak } = {} }) {
+  return !!speak;
+}
+
+function useSendBoxDictationStarted(dictateState) {
```

Comment:
Why are you returning the boolean value in an an array? Seems unnecessary if you're just deconstructing a single value.

Reply:
Do you mean this line?

```
const [activities] = useActivities();
```

`activities` is a property. This is for having a unified pattern for getter/setter pattern. Since we do not support setting properties, it looks this way.

For example, some properties support setting, e.g. suggested actions, it will looks like:

```
const [suggestedActions, setSuggestedActions] = useSuggestedActions();
```

Reply:
Oooooh... that makes a lot of sense.....